### PR TITLE
TUR-1974: Additional timeout logging

### PR DESCRIPTION
**Pull request by smitp33**
Adds more logging and a missing crawler

### CODE REVIEW

File: src/rendertron.ts [hunk 1]
```diff
@@ -152,14 +153,33 @@ export class Rendertron {
       return;
     }
 
+    const renderId = uuidv4()
     const started = performance.now();
 
+    const userAgent = ctx.request.headers['user-agent'] || undefined;
```
Comment (wwqrd) [inline src/rendertron.ts:159]:
```suggestion
    const userAgent = ctx.request.headers?.['user-agent'];
```

File: src/rendertron.ts [hunk 2]
```diff
@@ -152,14 +153,33 @@ export class Rendertron {
...
+    const renderUrl = ctx.url.replace('/render/','')
+    const marketMatch = renderUrl.match(marketRegex)
+    const market = marketMatch ? marketMatch[1] : "Not Found"
+    const renderCrawler =  userAgent ? detectCrawler(userAgent) : "Not Set";
+    const renderUserAgent =  userAgent || "Not Set";
```
Comment (wwqrd) [inline src/rendertron.ts:164]:
```suggestion
    const renderUserAgent =  userAgent ?? "Not Set";
```

File: src/rendertron.ts [hunk 3]
```diff
@@ -168,29 +188,29 @@ export class Rendertron {
...
     ctx.set('x-renderer', 'rendertron');
+    ctx.set('x-render-id', renderId);
     // Add custom headers to the response like 'Location'
     serialized.customHeaders.forEach((value: string, key: string) =>
       ctx.set(key, value)
     );
     ctx.status = serialized.status;
     ctx.body = serialized.content;
 
+    const timeout = !!serialized.customHeaders.get('timeout');
```
Comment (wwqrd) [inline src/rendertron.ts:199]:
```suggestion
    const timeout = Boolean(serialized.customHeaders.get('timeout'));
```

**Review by wwqrd — APPROVED** [review]
Not actually requesting these changes, but thought you might be interested in this more modern style.

**Review by JCorpe26 — APPROVED** [review]

**Pull request merged**